lib/elixir/lib/option_parser.ex

```diff
@@ -191,7 +194,7 @@ defmodule OptionParser do
       {[unlock: "path/to/file", unlock: "path/to/another/file"], [], []}
 
   """
-  @spec parse(argv, options) :: {parsed, argv, errors}
+  @spec parse(argv, options) :: {parsed, argv, errors} | no_return
```

Comment:
this is a bit confusing.
I was using this rationale, https://github.com/elixir-lang/elixir/pull/5391#issuecomment-257398494
which worked fine for me.

I was excluding `FunctionClauseError` out of the equation...
so by definition, a function with a guard could always return `no_return()` (sic)